core/storage/question/gae_models.py

```diff
@@ -357,59 +354,97 @@ def get_question_skill_links_based_on_difficulty_equidistributed_by_skill(
 
         question_skill_link_mapping = {}
 
+        # For fetching the questions randomly we have used a random offset.
+        # But this is a temporary solution since this method scales linearly.
+        # Other alternative methods were:
+        # 1) Using a random id in question id filter
+        # 2) Adding an additional column that can be filtered upon.
+        # But these methods are not viable because google datastore limit
```

Comment:
... limits each query to have at most one inequality filter. ...

(Great comment, btw!)

Reply:
Done.